### PR TITLE
[WFLY-15524]: Upgrade netty from 4.1.68 to 4.1.70.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Upgrading netty to 4.1.70.
* Removing netty-all and using the proper netty artefacts in pom.xml.
* Removing the io.netty jboss module and creating io.netty.netty-*
  modules for each artefact.
* Adding a task to manage the native parts.

Jira: https://issues.redhat.com/projects/WFLY/issues/WFLY-15524
Requires: https://github.com/wildfly/wildfly/pull/14886

### CODE REVIEW

File: ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-epoll/pm/wildfly/tasks.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    <copy-artifact artifact="io.netty:netty-transport-native-epoll::linux-x86_64" to-location="modules/system/layers/base/io/netty/netty-transport-native-epoll/main/lib" extract="true" >
+        <filter pattern="META-INF/" include="true" />
+        <filter pattern="META-INF/native/*" include="true" />
+        <filter pattern="*" include="false"/>
+    </copy-artifact>
+<!--    <copy-artifact artifact="io.netty:netty-transport-native-epoll::linux-aarch_64" to-location="modules/system/layers/base/io/netty/netty-transport-native-epoll/main/lib" extract="true" >
+        <filter pattern="META-INF/" include="true" />
+        <filter pattern="META-INF/native/*" include="true" />
+        <filter pattern="*" include="false"/>
+    </copy-artifact>-->
```
Comment (bstansberry) [inline ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-epoll/pm/wildfly/tasks.xml:12]: @ehsavoie Looks like this is pure cruft as it seems to be a commented out copy of the uncommented block.

Comment (ehsavoie) [reply ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-epoll/pm/wildfly/tasks.xml:12]: different classifier : linux-aarch_64

**Comment by bstansberry** [comment]
@pferraro @kabir Please review

File: ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-epoll/pm/wildfly/tasks.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" ?>
+<tasks xmlns="urn:wildfly:wildfly-feature-pack-tasks:2.0">
+    <copy-artifact artifact="io.netty:netty-transport-native-epoll::linux-x86_64" to-location="modules/system/layers/base/io/netty/netty-transport-native-epoll/main/lib" extract="true" >
+        <filter pattern="META-INF/" include="true" />
+        <filter pattern="META-INF/native/*" include="true" />
+        <filter pattern="*" include="false"/>
+    </copy-artifact>
+<!--    <copy-artifact artifact="io.netty:netty-transport-native-epoll::linux-aarch_64" to-location="modules/system/layers/base/io/netty/netty-transport-native-epoll/main/lib" extract="true" >
```
Comment (kabir) [inline ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-epoll/pm/wildfly/tasks.xml:8]: Cruft?

File: ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-unix-common/pm/wildfly/tasks.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" ?>
+<tasks xmlns="urn:wildfly:wildfly-feature-pack-tasks:2.0">
+    <copy-artifact artifact="io.netty:netty-transport-native-unix-common::linux-x86_64" to-location="modules/system/layers/base/io/netty/netty-transport-native-unix-common/main/lib" extract="true" >
+        <filter pattern="META-INF/" include="true" />
+        <filter pattern="META-INF/native/*" include="true" />
+        <filter pattern="*" include="false"/>
+    </copy-artifact>
+    <!--
+    <copy-artifact artifact="io.netty:netty-transport-native-epoll::linux-aarch_64" to-location="modules/system/layers/base/io/netty/netty-transport-native-epoll/main/lib" extract="true" >
```
Comment (kabir) [inline ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-unix-common/pm/wildfly/tasks.xml:9]: Cruft?

File: microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/vertx/core/main/module.xml [hunk 4]
```diff
@@ -33,7 +33,21 @@
     <dependencies>
         <module name="com.fasterxml.jackson.core.jackson-core"/>
         <module name="com.fasterxml.jackson.core.jackson-databind"/>
-        <module name="io.netty"/>
+        <!--<module name="io.netty"/>-->
```
Comment (kabir) [inline microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/vertx/core/main/module.xml:36]: Cruft?

**Review by kabir — DISMISSED** [review]
Some small comments aside, it looks good to me. I've restarted the two jobs that were failing.
**Linux SM JDK8** - had a failure in IIOP probably not related
**WildFly Preview** - had a lot of failures, hopefully just one of those bad runs

File: microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/netty/netty-resolver-dns/main/module.xml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        <property name="jboss.api" value="private"/>
+    </properties>
+
+    <resources>
+        <artifact name="${io.netty:netty-resolver-dns}"/>
+    </resources>
+
+    <dependencies>
+        <module name="java.logging"/>
+        <module name="java.management"/>
```
Comment (boris-unckel) [inline microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/netty/netty-resolver-dns/main/module.xml:36]: Wondering why I did not see these module.xml in the first round checking. Only java.naming is needed here.
java.logging, java.management can be removed.

File: microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/netty/netty-resolver-dns/main/module.xml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        <module name="java.naming"/>
+        <module name="java.xml"/>
+        <!--WFLY-14219 Remove deprecated <module name="javax.api"/> -->
+        <module name="sun.jdk"/>
+        <module name="jdk.sctp"/>
+        <module name="jdk.unsupported"/>
```
Comment (boris-unckel) [inline microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/netty/netty-resolver-dns/main/module.xml:42]: java.xml, sun.jdk, jdk.sctp, jdk.unsupported can be removed.

File: microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/netty/netty-codec-http2/main/module.xml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    <dependencies>
+        <module name="java.logging"/>
+        <module name="java.management"/>
+        <module name="java.naming"/>
+        <module name="java.xml"/>
+        <!--WFLY-14219 Remove deprecated <module name="javax.api"/> -->
+        <module name="sun.jdk"/>
+        <module name="jdk.sctp"/>
+        <module name="jdk.unsupported"/>
```
Comment (boris-unckel) [inline microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/netty/netty-codec-http2/main/module.xml:42]: java.logging, java.management, java.naming, java.xml, sun.jdk, jdk.sctp, jdk.unsupported can be removed

**Comment by kabir** [comment]
@wildfly/prod test

**Comment by kabir** [comment]
@ehsavoie the WF Preview failures seem related

File: microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/vertx/core/main/module.xml (hunk 4 above)
Comment (ehsavoie) [reply microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/vertx/core/main/module.xml:36]: yes

File: ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-unix-common/pm/wildfly/tasks.xml (hunk 3 above)
Comment (ehsavoie) [reply ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-unix-common/pm/wildfly/tasks.xml:9]: nope i was wondering about adding aarch_64 binaries

**Review by pferraro — APPROVED** [review]

**Comment by darranl** [comment]
@ehsavoie FYI this one has a merge conflict.

**Comment by ehsavoie** [comment]
@darranl should be ok now

**Comment by darranl** [comment]
@kabir could you please have another look, can we dismiss your review?

**Comment by kabir** [comment]
@darranl done

**Comment by bstansberry** [comment]
Thanks, @ehsavoie

**Pull request merged**